pkg/compact/compact_test.go

```diff
@@ -174,3 +178,360 @@ func BenchmarkGatherNoCompactionMarkFilter_Filter(b *testing.B) {
 	}
 
 }
+
+type planResult struct {
+	compactionBlocks, compactionRuns float64
+}
+
+func TestCompactProgressCalculate(t *testing.T) {
+	reg := prometheus.NewRegistry()
+	unRegisterer := &receive.UnRegisterer{Registerer: reg}
+	planner := NewTSDBBasedPlanner(log.NewNopLogger(), []int64{})
+
+	extLabels := labels.FromMap(map[string]string{"a": "1"})
+
+	for _, tcase := range []struct {
+		testName          string
+		testPlannerRanges []int64
+		input             []*metadata.Meta
+		expected          planResult
+	}{
+		// In this test case, the first four blocks are planned for compaction in the first run. These are then removed from the group and then the next two blocks from the original group are planned for compaction in the second run.
+		// Hence, a total of 6 blocks are planned for compaction over 2 runs.
+		{
+			testName: "two_runs",
+			testPlannerRanges: []int64{
```

Comment:
You don't have to make ranges a variable. Just passing them to the planner at line 189 is fine.

Reply:
The two ranges have different values. How can I pass them to planner in the beginning?

Reply:
Ok I see. But why do you need to use two different ranges here? Using one range up to 2d is the same for the two, right?

Reply:
Alright, addressed this in a recent commit.